src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -207,6 +218,9 @@ private DetailAST getAncestorAst(TreeWalkerAuditEvent event) {
         DetailAST eventAst;
         if (containsShortName(checkNamesForContextStrategyByTokenOrAncestorSet, event)) {
             eventAst = getEventAst(event).getParent();
+            while (!ancestorTokenSet.contains(eventAst.getType())) {
```

Comment:
please add check for null, because it can fail if user sets some unexpected Check to checkNamesForContextStrategyByTokenOrAncestorSet.
`while (eventAst != null && !ancestorTokenSet.contains(eventAst.getType())) {`

Reply:
done